features/mbedtls/platform/inc/mbedtls_config_reader.h

```diff
@@ -0,0 +1,500 @@
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDHE_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_PKCS1_V21
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+    #define MBEDTLS_RSA_C
+    
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+    #define MBEDTLS_ASN1_PARSE_C
+    #define MBEDTLS_PK_PARSE_C
+    #define MBEDTLS_PK_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+#endif
+
+#define MBED_TLS_HAVE_ASM
+#define MBED_TLS_HAVE_TIME
+#define MBEDTLS_HAVE_TIME
+#define MBEDTLS_REMOVE_ARC4_CIPHERSUITES
+#define MBEDTLS_CIPHER_C
+#define MBEDTLS_ERROR_STRERROR_DUMMY
+#define MBEDTLS_X509_CHECK_KEY_USAGE
+#define MBEDTLS_X509_CHECK_EXTENDED_KEY_USAGE
+#define MBEDTLS_VERSION_C
+#define MBEDTLS_SELF_TEST
+#define MBEDTLS_PK_RSA_ALT_SUPPORT
+#define MBEDTLS_SSL_SESSION_TICKETS
+#define MBEDTLS_VERSION_FEATURES
+#define MBEDTLS_DEBUG_C
+#define MBEDTLS_ERROR_C
+#define MBEDTLS_AES_C
+#define MBEDTLS_SHA256_C
+
+#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT) || defined(MBEDTLS_TEST_NULL_ENTROPY) || \
+    defined(MBEDTLS_ENTROPY_NV_SEED)
+
+#define MBEDTLS_HMAC_DRBG_C
+#define MBEDTLS_ENTROPY_C
+#define MBEDTLS_CTR_DRBG_C
+#define MBEDTLS_NO_PLATFORM_ENTROPY
+#define MBEDTLS_SSL_ALL_ALERT_MESSAGES
+#define MBEDTLS_SSL_ALPN
+#define MBEDTLS_SSL_CACHE_C
+#define MBEDTLS_SSL_COOKIE_C
+#define MBEDTLS_SSL_CLI_C
+#define MBEDTLS_SSL_DTLS_ANTI_REPLAY
+#define MBEDTLS_SSL_DTLS_BADMAC_LIMIT
+#define MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
+#define MBEDTLS_SSL_DTLS_HELLO_VERIFY
+#define MBEDTLS_SSL_ENCRYPT_THEN_MAC
+#define MBEDTLS_SSL_EXPORT_KEYS
+#define MBEDTLS_SSL_EXTENDED_MASTER_SECRET
+#define MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
+#define MBEDTLS_SSL_PROTO_DTLS
+#define MBEDTLS_SSL_PROTO_TLS1_2
+#define MBEDTLS_SSL_RENEGOTIATION
+#define MBEDTLS_SSL_TLS_C
+
+#endif // #if defined(MBEDTLS_ENTROPY_HARDWARE_ALT) || defined(MBEDTLS_TEST_NULL_ENTROPY) || defined(MBEDTLS_ENTROPY_NV_SEED)
```

Comment:
Style: Normally in Mbed TLS we drop the `define`, redundant brackets and the `#if`.

Reply:
This is a holdover from when it was an if/else.
Will remove